common/auth/tls_config_helper_test.go

```diff
@@ -0,0 +1,334 @@
+
+// test if the input is valid
+func Test_NewTLSConfig(t *testing.T) {
+	tests := map[string]struct {
+		cfg    *TLS
+		cfgErr string
+	}{
+		"emptyConfig": {
+			cfg: &TLS{},
+		},
+		"caData_good": {
+			cfg: &TLS{
+				Enabled: true,
+				CaData:  validBase64CaData,
+			},
+		},
+		"caData_badBase64": {
+			cfg:    &TLS{Enabled: true, CaData: "this isn't base64"},
+			cfgErr: "illegal base64 data at input byte",
+		},
+		"caData_badPEM": {
+			cfg:    &TLS{Enabled: true, CaData: "dGhpcyBpc24ndCBhIFBFTSBjZXJ0"},
+			cfgErr: "unable to parse certs as PEM",
+		},
+		"clientCert_badbase64cert": {
+			cfg: &TLS{
+				Enabled:  true,
+				CertData: "this ain't base64",
+				KeyData:  validBase64Key,
+			},
+			cfgErr: "illegal base64 data at input byte",
+		},
+		"clientCert_badbase64key": {
+			cfg: &TLS{
+				Enabled:  true,
+				CertData: validBase64Certificate,
+				KeyData:  "this ain't base64",
+			},
+			cfgErr: "illegal base64 data at input byte",
+		},
+		"clientCert_missingprivatekey": {
+			cfg: &TLS{
+				Enabled:  true,
+				CertData: validBase64Certificate,
+				KeyData:  "",
+			},
+			cfgErr: "unable to config TLS: cert or key is missing",
+		},
+		"clientCert_duplicate_cert": {
+			cfg: &TLS{
+				Enabled:  true,
+				CertData: validBase64Certificate,
+				CertFile: "/a/b/c",
+			},
+			cfgErr: "only one of certData or certFile properties should be specified",
+		},
+		"clientCert_duplicate_key": {
+			cfg: &TLS{
+				Enabled: true,
+				KeyData: validBase64Key,
+				KeyFile: "/a/b/c",
+			},
+			cfgErr: "only one of keyData or keyFile properties should be specified",
+		},
+		"clientCert_duplicate_ca": {
+			cfg: &TLS{
+				Enabled: true,
+				CaData:  validBase64CaData,
+				CaFile:  "/a/b/c",
+			},
+			cfgErr: "only one of caData or caFile properties should be specified",
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			_, err := NewTLSConfig(tc.cfg)
+			if tc.cfgErr != "" {
+				assert.ErrorContains(t, err, tc.cfgErr)
+			} else {
+				assert.NoErr(t, err)
```

Comment:
Ugh, build error, it's actually this.

```suggestion
				assert.NoError(t, err)
```

Reply:
yeah. I noticed that and fixed in last commit. Thank you for supporting